server/netty-server/src/main/scala/sttp/tapir/server/netty/internal/reactivestreams/FileWriterSubscriber.scala

```diff
@@ -0,0 +1,61 @@
+package sttp.tapir.server.netty.internal.reactivestreams
+
+import io.netty.handler.codec.http.HttpContent
+import org.reactivestreams.{Publisher, Subscription}
+
+import java.nio.channels.AsynchronousFileChannel
+import java.nio.file.{Path, StandardOpenOption}
+import scala.concurrent.{Future, Promise}
+
+class FileWriterSubscriber(path: Path) extends PromisingSubscriber[Unit, HttpContent] {
+  private var subscription: Subscription = _
+  private var fileChannel: AsynchronousFileChannel = _
+  private var position: Long = 0
+  private val resultPromise = Promise[Unit]()
+
+  override def future: Future[Unit] = resultPromise.future
+
+  override def onSubscribe(s: Subscription): Unit = {
+    this.subscription = s
+    fileChannel = AsynchronousFileChannel.open(path, StandardOpenOption.WRITE, StandardOpenOption.CREATE)
+    s.request(1)
+  }
+
+  override def onNext(httpContent: HttpContent): Unit = {
+    val byteBuffer = httpContent.content().nioBuffer()
+    fileChannel.write(
+      byteBuffer,
+      position,
+      (),
+      new java.nio.channels.CompletionHandler[Integer, Unit] {
+        override def completed(result: Integer, attachment: Unit): Unit = {
+          position += result
+          subscription.request(1)
+        }
+
+        override def failed(exc: Throwable, attachment: Unit): Unit = {
+          subscription.cancel()
+          onError(exc)
+        }
+      }
+    )
+  }
+
+  override def onError(t: Throwable): Unit = {
+    fileChannel.close()
```

Comment:
shouldn't we delete the file on error or sth? or maybe the raw value mechanism already handles that

Reply:
In `ServerInterpreter.decodeBody` there's a general mechanism that deletes created files in case of decoding failures. When I change `toRaw` to return a `F[DecodeResult[RawValue[R]]]`, that handling may be extended to delete also on `toRaw` failures, right?
Additionally, we may want to try deleting files also on failing `F[]`, just in case something unexpected happens there, wdyt?

Reply:
yeah we should also delete on errors ... but I'm surprised it's not already happening. Maybe it is, but at a different place? ;)

Reply:
but maybe a failed `F` is considered a bug in tapir 🤔

Reply:
(which would make sense - any errors that we want to handle are then represented as a value - here, `DecodeResult`)

Reply:
Makes sense, it is not clear when one can fail `F`. I think it's because `sttp.monad.MonadError` is often used only as "monad capability", which isn't extracted to its own typeclass.